pkgs/top-level/all-packages.nix

```diff
@@ -20193,7 +20193,11 @@ with pkgs;
   glibcInfo = callPackage ../development/libraries/glibc/info.nix { };
 
   glibc_multi = callPackage ../development/libraries/glibc/multi.nix {
-    glibc32 = pkgsi686Linux.glibc;
+    # The buildPackages is required for cross-compilation. The pkgsi686Linux set
+    # has target and host always set to the same value based on target platform
+    # of the current set. We need host to be same as build to correctly get i686
+    # variant of glibc.
+    glibc32 = pkgsi686Linux.buildPackages.glibc;
```

Comment:
Not sure if `buildPackages.pkgsi686Linux.glibc` would work, but if so maybe less confusing, because it's then analogous to any other case where we take stuff from buildPackages.

Reply:
Unfortunately, it won't work due to the issue described here: https://github.com/NixOS/nixpkgs/issues/212494#issuecomment-1404833636. The `buildPackages` is set with the build, and host platform is the same, and the target platform is cross-platform. The issue is that `pkgsi686Linux` set gets generated with the host and target set to the same platform, no matter the real host platform of the original set. It is limited in nixpkgs where it takes only `localSystem` and `crossSystem`; there is no way to specify "targetSystem" and thus no way to create a set where host and build are the same while target is different. The result is that this suggestion won't work.